src/AggregateFunctions/AggregateFunctionDeltaSum.h

```diff
@@ -87,16 +85,15 @@ class AggregationFunctionDeltaSum final
             place_data->sum += rhs_data->sum;
             place_data->first = rhs_data->first;
```

Comment:
BTW, it seems to be incorrect process of counter reset case. Why do we compare rhs_data->last < place_data->first instead of place_data->last > rhs_data->first? Why do we set place_data->first = rhs_data->first instead of place_data->last = rhs_data->?
Lets see an example:
If we have such sequences: [4, 6] [1, 2] [5, 8]. States are (first=4, last=6, sum=2), (first=1, last=2, sum=1), (first=5, last=8, sum=3).
First merge: merge((first=4, last=6, sum=2), (first=1, last=2, sum=1)) = (first=1, last=6, sum=3), but this state should represent [4, 6, 1, 2] sequence.
Second merge: merge((first=1, last=6, sum=2), (first=5, last=8, sum=3)) = |this case you don't process as count reset, because check 8 < 1 failed and we just skip left result here| = (first=5, last=8, sum=3).

Or I missed something?

Reply:
(Ugh, sorry. I'd had a response to this comment but I forgot to hit "add single comment" so it was in the review which I never submitted.)

Yeah, I think you're right, that is a better comparison to be done here.

Should I fix this in this PR or a separate one?

Reply:
> Should I fix this in this PR or a separate one?

I think it would be better to fix in a separate PR with bugfix category to have an ability to backport it

Reply:
Should I also move the other small improvement to a different PR as well?

Reply:
We need the new PR only for backporting fixing possible incorrect behaviour, the other small improvement we can leave here

Reply:
https://github.com/ClickHouse/ClickHouse/pull/23437